### PR TITLE
Finished background , platform and "camera"

### PR DESCRIPTION
**Pull request by DaniloMatic99**
Finished background and platform, and also something that act like camera where player is fixed.

### CODE REVIEW

File: src/Main.cpp [hunk 1]
```diff
@@ -41,71 +46,164 @@ struct PlayerControlsSystem
...
-					sprite.change_to("fallingHero/attack");
-				if(side == false)
-					sprite.change_to("fallingHero/attackLeft");
-			}
-			else{
-				if(side == true)
-					sprite.change_to("fallingHero/heroStanding");
-				if(side == false)
-					sprite.change_to("fallingHero/leftHeroStanding");
+					// if (keys.is_down(bindings.up))
```
Comment (gavrilovmiroslav) [inline src/Main.cpp:51]: Indentation is all over the place, please pay attention to this and match the rest of the engine.

File: src/Main.cpp [hunk 2]
```diff
@@ -41,71 +46,164 @@ struct PlayerControlsSystem
...
+							pos.xy.x = SCREEN_WIDTH - 157;
+					}
+					else if (keys.is_down(bindings.attack))
+					{
+						if(side == true)
+							sprite.change_to("fallingHero/attack");
+						if(side == false)
+							sprite.change_to("fallingHero/attackLeft");
+					}
+					else{
```
Comment (gavrilovmiroslav) [inline src/Main.cpp:84]: Braces always go onto their own line and are not additionally indented.

File: src/Main.cpp [hunk 3]
```diff
@@ -41,71 +46,164 @@ struct PlayerControlsSystem
...
+					{
+						side = true;
+						sprite.change_to("fallingHero/runningRight");
+						pos.xy.x += SPEED_MOD * Time::delta_time();
+						if(pos.xy.x > SCREEN_WIDTH - 157)
+							pos.xy.x = SCREEN_WIDTH - 157;
+					}
+					else if (keys.is_down(bindings.attack))
+					{
+						if(side == true)
```
Comment (gavrilovmiroslav) [inline src/Main.cpp:79]: Please don't check `x == true`, just check `x` if it's a boolean.

File: src/Main.cpp [hunk 4]
```diff
@@ -41,71 +46,164 @@ struct PlayerControlsSystem
...
+	, public MutAccessGroupStorage<Background,KeyBindings, Position, Sprite>
+{
+	int counter = 0;
+
+	void on_tick() override
+	{
+		const auto& keys = KeyState::get();
+		
+			for (auto&& [background_entity, bindings, background_pos, sprite] :access_storage().each())
+				{
```
Comment (gavrilovmiroslav) [inline src/Main.cpp:107]: This whole method needs to be reindented carefully.

File: src/Rendering.cpp [hunk 5]
```diff
@@ -2,7 +2,7 @@
 #include "Access.h"
 #include "FPNG.h"
 
-#include <SDL.h>
+#include <SDL2/SDL.h>
```
Comment (gavrilovmiroslav) [inline src/Rendering.cpp:5]: This should now be fixed and actually incorrect - pull from main and compile from meson and VSCode

**Review by gavrilovmiroslav — DISMISSED** [review]

**Review by msimic01 — APPROVED** [review]

**Review by msimic01 — APPROVED** [review]

**Review by gavrilovmiroslav — APPROVED** [review]
Generally you don't want to let your review to grow to this size, but let's do it! Chunkification is key!

**Comment by gavrilovmiroslav** [comment]
Are we closing this, @DaniloMatic99 ?

**Pull request merged**